src/crimson/osd/shard_services.cc

```diff
@@ -802,8 +802,9 @@ seastar::future<> ShardServices::dispatch_context_messages(
 
 seastar::future<> ShardServices::dispatch_context(
   crimson::os::CollectionRef col,
-  PeeringCtx &&ctx)
+  PeeringCtx &&pctx)
 {
+  auto ctx = std::move(pctx);
```

Comment:
I'm not sure I'm understanding this change. Can you please explain?
Looking at `dispatch_context_transaction` (from the backtrace), it seems that we should be passing rval (`PeeringCtx &&ctx`) as parameter and not as a reference, right? Could that explain the tracker?

Reply:
Yeah, but not enough, I've just updated this PR.
Passing rval as parameter can solve this issue if `ShardServices::dispatch_context()` is also a coroutine. But in this case, it's not, so I used `seastar::do_with()` to hold `PeeringCtx` in the latest update. Please take a look.